Add BBCode (NGA)

stone5265 opened the pull request:
<!--
The manual review may take several days or weeks,
depending on the reviewer's availability and workload.
Patience padawan!

You can request a review from @packagecontrol-bot.
Please ensure the reviews pass and follow any instructions.

Please provide some information via this checklist,
feel free to remove what't not applicable.
-->
- [X] I'm the package's author and/or maintainer.
- [X] I have have read [the docs][1].
- [X] I have tagged a release with a [semver][2] version number.
- [X] My package repo has a description and a README describing what it's for and how to use it.
- [ ] My package doesn't add context menu entries. *
- [ ] My package doesn't add key bindings. **
- [ ] Any commands are available via the command palette.
- [ ] Preferences and keybindings (if any) are listed in the menu and the command palette, and open in split view.
- [X] If my package is a syntax it doesn't also add a color scheme. ***
- [X] If my package is a syntax it is named after the language it supports (without suffixes like "syntax" or "highlighting").
- [X] I use [.gitattributes][3] to exclude files from the package: images, test files, sublime-project/workspace.

My package is A Sublime Text plugin supporting BBCode (NGA) syntax, including syntax highlighting, snippets, and other features. In which, NGA is a BBS forum, fully known as 艾泽拉斯国家地理 (National Geographic of Azeroth).

There are no packages like it in Package Control.
| Syntax                      | Snippets                       | Key Bindings & Context Menu          |
| :---------------------: | :-----------------------: | :-----------------------: |
| ![syntax1][syntax1_gif] | ![snippets][snippets_gif] | ![commands][commands_gif] |
| ![syntax2][syntax2_gif] |                                    |                                          |


<!-- 
*)   Unless it definitely really needs them,
     they apply to the cursor's context
     and their visibility is conditional.
     Space in this menu is limited!
**)  There aren't enough keys for all packages,
     you'd risk overriding those of other packages.
     You can put commented out suggestions in a keymap file, 
     and/or explain how to create bindings in your README.
***) We have hundreds of color schemes,
     and plenty of scopes to make any syntax work. 

For bonus points also considered how the review guidelines apply to your package:
https://github.com/wbond/package_control_channel/wiki#reviewing-a-package-addition

For updates to existing packages:
If your package isn't using tag based releases,
please switch to tags now.
 -->

[1]: https://packagecontrol.io/docs/submitting_a_package
[2]: https://semver.org
[3]: https://www.git-scm.com/docs/gitattributes#_export_ignore
[commands_gif]: https://raw.githubusercontent.com/stone5265/Sublime-BBCode-NGA/refs/heads/master/figs/commands.gif
[snippets_gif]: https://raw.githubusercontent.com/stone5265/Sublime-BBCode-NGA/refs/heads/master/figs/snippets.gif
[syntax2_gif]: https://raw.githubusercontent.com/stone5265/Sublime-BBCode-NGA/refs/heads/master/figs/syntax_error.gif
[syntax1_gif]: https://raw.githubusercontent.com/stone5265/Sublime-BBCode-NGA/refs/heads/master/figs/syntax_surroundings.gif

braver commented:
You'll have to remove the trailing `,` at `"1.0.2": "messages/1.0.2.txt",` for the JSON to be valid.

I notice you're careful about the context menu only being visible, and the event listeners only doing something, if applicable, so that's good. For the event listener, I wonder if you can optimize further using [is_applicable()](https://www.sublimetext.com/docs/api_reference.html#sublime_plugin.ViewEventListener.is_applicable)... or if that's even necessary. Anyway, mostly looks good! 👍🏻

stone5265 commented:
Thank you for your suggestions. I have corrected the `messages.json` and modified the `EventListener` to `ViewEventListener`, using `is_applicable()` to restrict the scope.

braver commented:
Oh, those first four messages in the messages.json won’t work Can you remove them? That’s the final sticking point for the PR.

stone5265 commented:
It removed.

stone5265 commented:
It's done. Sorry for the trouble, and thanks for your patience!
